client/src/controllers/PreviewController.ts

```diff
@@ -0,0 +1,789 @@
+import axe from 'axe-core';
+
+import { Controller } from '@hotwired/stimulus';
+import type { ContextObject } from 'axe-core';
+import {
+  getAxeConfiguration,
+  getA11yReport,
+  renderA11yResults,
+} from '../includes/a11y-result';
+import { wagtailPreviewPlugin } from '../includes/previewPlugin';
+import {
+  getPreviewContentMetrics,
+  renderContentMetrics,
+} from '../includes/contentMetrics';
+import { WAGTAIL_CONFIG } from '../config/wagtailConfig';
+import { debounce } from '../utils/debounce';
+import { gettext } from '../utils/gettext';
+import type { ProgressController } from './ProgressController';
+import { setOptionalInterval } from '../utils/interval';
+
```

Comment:
One quick thought, before a full review, should we move these checks functions outside (utils) the controller module?


Or maybe just make them function declarations so they can sit at the end of the file and just be hoisted up for usage. May help with readability of this file if these two functions are not at the top

Reply:
I believe Thibaud would like to do some further work on the checkers in this release, so I would expect those functions and their calls to be moved out of the `PreviewController` code entirely in favour of listening to the `w-preview:loaded` event.

I tried to leave them untouched until that happens. If desired I can move the function declarations outside this file for now, but still call them directly in the `replaceIframe` method as I think refactoring it to use the event is out of scope for this PR.

Reply:
Good call, let's leave this for now.